### ISSUE TITLE
[Feature] Capacitor 3 support

### ISSUE THREAD

**Issue by arielhernandezmusa** [nxtend-team/nxtend#475]
there is an intention to support capacitor 3?

**Comment by brian-weasner** [comment]
Not the maintainer, but it seems that we may be able to update capacitor versions by changing the version in this file.
https://github.com/nxtend-team/nxtend/blob/d0d25b16fa1fd4402832b17f297eaf4658aa83bd/packages/capacitor/src/utils/versions.ts#L1

Not sure if this will even work, or what it may all affect.

**Comment by Stevoohh** [comment]
Updated to version: ^3.0.1. Still not working for me. 
Any news on this?

**Comment by devinshoemaker** [comment]
A major update to all the plugins is in progress. Developing and testing these plugins is pretty time consuming and I work on these in my free time so I don't have an ETA, but I can tell you the work is being done.

**Comment by hellokingdom** [comment]
@devinshoemaker thank you for all the hard work. Would you recommend for a new project to work with Cap 2 before migrating in future?

**Comment by brian-weasner** [comment]
Seeing that the plugin uses whatever capacitor version is installed in node modules, I was able to upgrade to capacitor v3.1.2 by updating all my @capacitor npm packages.

Updated in both the root of my monorepo & within my mobile app folder.

For me these were in the monorepo's `package.json`
- @capacitor/core
- @capacitor/android
- @capacitor/ios

And this was in the mobile app's `package.json`
- @capacitor/cli

Follow this guide from Capacitor for the rest of the upgrade process from v2 to v3
https://capacitorjs.com/docs/updating/3-0

**Comment by devinshoemaker** [comment]
Thanks @brian-weasner, that is correct. It should be very easy to generate a Capacitor app with this plugin and then immediately migrate to Capacitor 3. I've been hard at work on some major updates, so rest assured that Capacitor 3 support is coming down the pipeline.

**Comment by brian-weasner** [comment]
@DeniKucevic I am using v3.1.2 as well, I am able to get the plugins working.
If I press the back button on the first/initial page it closes the app, however on other pages it does not close the app, it navigates back.
As far as plugins go, they work for me, just make sure you have them in the monorepo's base `package.json` and the app's `package.json`. Make sure you did the `npm install` in both places, and make sure that after you run the `nx run app-name:sync` they show up in the `capacitor.settings.gradle` file (for android).

**Comment by DeniKucevic** [comment]
hmm.. I am not getting that behavior. No matter on what page I am it closes the app. I tried getting the App from '@capacitor/app' to listen to the back button press but all I get is 'app plugin is not implemented on android' error in android studio. I tried having the empty project and all works well until I update the capacitor... I am 100% making a mistake in updating it then, I just don't see what yet.
Thank you for the response!

**Comment by brian-weasner** [comment]
@DeniKucevic No problem, I just followed https://capacitorjs.com/docs/updating/3-0 step by step and it worked for me. (I'm using the exact versions of gradle that they suggest in the upgrade guide too, although I'm not sure it makes a difference)

**Comment by brian-weasner** [comment]
@DeniKucevic 
I did some more digging, in my app we've always had some code similar to below
```ts
App.addListener('backButton', (backButtonEvent) => {
      if (backButtonEvent.canGoBack) {
        this.location.back();
      }
});
```
So if you are having issues with @capacitor/app not showing up/being available, its possible it's because you added it to your monorepo's `package.json` but not your capacitor-app's `package.json`. (As noted here: https://nxtend.dev/docs/capacitor/getting-started#adding-capacitor-plugins)

After having it in both places and then running the `nx run capacitor-app-name:sync:android` it should show `@capacitor/app` within the found plugins list.
![image](https://user-images.githubusercontent.com/84519320/132768933-aae2f172-e712-49a5-a0dc-e8b0b5a3f7ae.png)

**Comment by DeniKucevic** [comment]
@brian-weasner Yes, the problem for me was that I did not have plugins in both project and monorepo! I fugured it out earlier today from your comment 

> As far as plugins go, they work for me, just make sure you have them in the monorepo's base package.json and the app's package.json.

Thank you man, I really enjoy working in the nx and this would have been a big deal for me 👍

**Comment by spicemix** [comment]
Is there any way to get a live-reload running in Cap 3 with @nxtend/capacitor? Thanks!

```shell
ionic capacitor run ios -l --external
```

**Comment by brian-weasner** [comment]
@spicemix It looks like Devin has mentioned how to get live reload working in another issue (#459) I would look here
https://github.com/nxtend-team/nxtend/issues/459#issuecomment-804497041

**Comment by pareshgami** [comment]
@brian-weasner Thanks. It's working till the date. Adding app plugin into both the package.json. One root level and another inside at application level

**Comment by spicemix** [comment]
Just FYI, this project has quietly been replaced by nxext. That and the ionic extension for VS code are good places to start looking now.

Devin, I would update the docs on this project accordingly, to redirect people to the new code. Thanks for your work on this and I was thrilled to see it's still going.

**Comment by brian-weasner** [comment]
@spicemix I don't see any reference to this project from nxext, and it's not fork from this project. I also only see @devinshoemaker listed as a contributor.

I'd much rather stick with nxtend because I know this is maintained by Devin (an actual employee at Ionic, which is the company that makes Capacitor) rather than go with nxext which seems to be a copy of all plugins from this project without having this project as an upstream.

**Comment by ProfEibe** [comment]
@devinshoemaker said it himself that he will quit this project and it will relive in the other one.

**Comment by brian-weasner** [comment]
Found where he mentioned that. Linking for reference.
https://github.com/nxtend-team/nxtend/pull/638#issuecomment-1204196754
